controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,14 +134,335 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to create StorageClass: %v", err)
+			}
+			err = r.setRookCSICephFS(enableRookCSICephFS)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to set RookEnableCephFSCSIKey to EnableRookCSICephFS [%v]: %s", enableRookCSICephFS, err)
+			}
+			return reconcile.Result{}, nil
+		}
+
+		// Update the StorageClassClaim status.
+		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimReady
+
+	}
+	// TODO(@rexagod): Deletion flow for SCC on Consumer side
+	// Close client-side connections.
+	providerClient.Close()
+
 	return reconcile.Result{}, nil
 }
 
-func (r *StorageClassClaimReconciler) reconcileProviderPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
-	log := ctrllog.FromContext(ctx)
+func (r *StorageClassClaimReconciler) setRookCSICephFS(enableRookCSICephFS bool) error {
+	rookCephOperatorConfig := &corev1.ConfigMap{}
+	err := r.get(rookCephOperatorConfig)
+	if err != nil {
+		r.log.Error(err, "Unable to get RookCeph ConfigMap.", "RookCephConfigMap", klog.KRef(r.storageCluster.Namespace, "rook-ceph-operator-config"))
+		return err
+	}
+	enableDisableFlagStr := fmt.Sprintf("%v", enableRookCSICephFS)
+
+	// If the current state of 'ROOK_CSI_ENABLE_CEPHFS' flag is same, just return.
+	if rookCephOperatorConfig.Data["ROOK_CSI_ENABLE_CEPHFS"] == enableDisableFlagStr {
+		return nil
+	}
+	rookCephOperatorConfig.Data["ROOK_CSI_ENABLE_CEPHFS"] = enableDisableFlagStr
+	return r.Client.Update(r.ctx, rookCephOperatorConfig)
+}
+
+func (r *StorageClassClaimReconciler) createStorageClasses(availableStorageClassConfigurations []StorageClassConfiguration, storageClassClaimOwnerReference metav1.OwnerReference) error {
+	var skippedStorageClass []string
+	for _, storageclassConfiguration := range availableStorageClassConfigurations {
+		if storageclassConfiguration.reconcileStrategy == "ignore" || storageclassConfiguration.disable {
+			continue
+		}
+
+		// StorageClass received in the response does not have the same name as the consumer-created StorageClassClaim.
+		storageClass := storageclassConfiguration.storageClass
+		if strings.Contains(storageClass.Name, "-ceph-rbd") && !storageclassConfiguration.isClusterExternal {
+
+			// Wait for CephBlockPool to be ready.
+			cephBlockPool := cephv1.CephBlockPool{}
+			err := r.get(&cephBlockPool)
+			if err != nil || cephBlockPool.Status == nil || cephBlockPool.Status.Phase != cephv1.ConditionType(util.PhaseReady) {
+				r.log.Info("Waiting for CephBlockPool to be Ready. Skip reconciling StorageClass", "CephBlockPool", klog.KRef(storageClass.Parameters["pool"], storageClass.Parameters["clusterID"]), "StorageClass", klog.KRef("", storageClass.Parameters["clusterID"]))
+				skippedStorageClass = append(skippedStorageClass, storageClass.Name)
+				continue
+			}
+		} else if strings.Contains(storageClass.Name, "-cephfs") && !storageclassConfiguration.isClusterExternal {
```

Comment:
Never try to deduce info from a name, this is a bad habit. We know the type of configuration on the call site so just pass it in.